python/ray/worker.py

```diff
@@ -165,6 +164,22 @@ def __init__(self):
         # increment every time when `ray.shutdown` is called.
         self._session_index = 0
 
+    @property
+    def connected(self):
+        return self.node is not None
+
+    @property
+    def node_ip_address(self):
+        if self.node is None:
```

Comment:
Use `self. connected()` instead?

Reply:
yeah, and better use a `check_connected` method. Because the code is duplicated in multiple places.
Also, `AttributeError` doesn't seems to be the proper exception type. I'd use `RuntimeError` in this case.